src/editor/CSSInlineEditor.js

```diff
@@ -164,24 +164,37 @@ define(function (require, exports, module) {
      */
     function htmlToCSSProvider(hostEditor, pos) {
 
-        // Only provide a CSS editor when cursor is in HTML content
-        if (hostEditor.getLanguageForSelection().getId() !== "html") {
+        function queryProvider(hostEditor, pos) {
+             // Only provide a CSS editor when cursor is in HTML content
+             if (hostEditor.getLanguageForSelection().getId() !== "html") {
+                 return null;
+             }
+
+             // Only provide CSS editor if the selection is within a single line
+             var sel = hostEditor.getSelection();
+             if (sel.start.line !== sel.end.line) {
+                 return null;
+             }
+
+             // Always use the selection start for determining selector name. The pos
+             // parameter is usually the selection end.
+             var selectorResult = _getSelectorName(hostEditor, sel.start);
+             if (selectorResult.selectorName === "") {
+                 return selectorResult.reason || null;
+             }
+             return selectorResult;
+         }
+        // so we can see this function outside
+        htmlToCSSProvider.queryProvider = queryProvider;
```

Comment:
Move this outside the function, so it doesn't get redone every time we call this function:

```js
function htmlToCSSProvider(hosteEditor, pos) {
...
}
htmlToCSSProvider.queryProvider = function(hostEditor, pos) {
...
};
```